MemorySanitizer: use-of-uninitialized-value in sqlite3BtreeOpen

From https://github.com/bitcoin/bitcoin/pull/21629/checks?check_run_id=2285803701 (#21629).

```bash
2021-04-07T10:27:19.893068Z [scheduler] [validationinterface.cpp:226] [operator()] BlockConnected: block hash=000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f blUninitialized bytes in __interceptor_strcmp at offset 0 inside [0x70100000051c, 1)
==82166==WARNING: MemorySanitizer: use-of-uninitialized-value
    #0 0x557a2f20eabc in sqlite3BtreeOpen (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/test_bitcoin+0x2a60abc)
    #1 0x557a2f2077b3 in openDatabase (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/test_bitcoin+0x2a597b3)
    #2 0x557a2edb3d65 in SQLiteDatabase::Open() /tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/wallet/sqlite.cpp:204:19
    #3 0x557a2edaee72 in SQLiteDatabase::SQLiteDatabase(boost::filesystem::path const&, boost::filesystem::path const&, bool) /tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/wallet/sqlite.cpp:65:9
    #4 0x557a2ed5e878 in std::__1::__unique_if<SQLiteDatabase>::__unique_single std::__1::make_unique<SQLiteDatabase, char const (&) [1], char const (&) [1], bool>(char const (&) [1], char const (&) [1], bool&&) /tmp/cirrus-ci-build/ci/scratch/msan/build/include/c++/v1/memory:3028:32
    #5 0x557a2ed4b40b in CreateMockWalletDatabase() /tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/wallet/walletdb.cpp:1099:12
    #6 0x557a2c934eb8 in WalletTestingSetup::WalletTestingSetup(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) /tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/wallet/test/wallet_test_fixture.cpp:9:40
    #7 0x557a2d7ba424 in psbt_wallet_tests::parse_hd_keypath::parse_hd_keypath() /tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/wallet/test/psbt_wallet_tests.cpp:77:1
    #8 0x557a2d7a6ab6 in psbt_wallet_tests::parse_hd_keypath_invoker() /tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/wallet/test/psbt_wallet_tests.cpp:77:1
    #9 0x557a2ca2e998 in boost::detail::function::void_function_invoker0<void (*)(), void>::invoke(boost::detail::function::function_buffer&) /tmp/cirrus-ci-build/depends/x86_64-pc-linux-gnu/include/boost/function/function_template.hpp:117:11
    #10 0x557a2f18f6cc in boost::function0<void>::operator()() const /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/function/function_template.hpp:763:14
    #11 0x557a2f18f6cc in boost::detail::forward::operator()() /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/execution_monitor.ipp:1368:32
    #12 0x557a2f18f6cc in boost::detail::function::function_obj_invoker0<boost::detail::forward, int>::invoke(boost::detail::function::function_buffer&) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/function/function_template.hpp:137:18
    #13 0x557a2f181edf in boost::function0<int>::operator()() const /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/function/function_template.hpp:763:14
    #14 0x557a2f181edf in int boost::detail::do_invoke<boost::shared_ptr<boost::detail::translator_holder_base>, boost::function<int ()> >(boost::shared_ptr<boost::detail::translator_holder_base> const&, boost::function<int ()> const&) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/execution_monitor.ipp:290:30
    #15 0x557a2f181edf in boost::execution_monitor::catch_signals(boost::function<int ()> const&) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/execution_monitor.ipp:879:16
    #16 0x557a2f1825ca in boost::execution_monitor::execute(boost::function<int ()> const&) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/execution_monitor.ipp:1277:16
    #17 0x557a2f186f29 in boost::execution_monitor::vexecute(boost::function<void ()> const&) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/execution_monitor.ipp:1377:5
    #18 0x557a2f09ac73 in boost::unit_test::unit_test_monitor_t::execute_and_translate(boost::function<void ()> const&, unsigned long) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/unit_test_monitor.ipp:49:9
    #19 0x557a2f00fb22 in boost::unit_test::framework::state::execute_test_tree(unsigned long, unsigned long, boost::unit_test::framework::state::random_generator_helper const*) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/framework.ipp:823:44
    #20 0x557a2f010d47 in boost::unit_test::framework::state::execute_test_tree(unsigned long, unsigned long, boost::unit_test::framework::state::random_generator_helper const*) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/framework.ipp:792:58
    #21 0x557a2f010d47 in boost::unit_test::framework::state::execute_test_tree(unsigned long, unsigned long, boost::unit_test::framework::state::random_generator_helper const*) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/framework.ipp:792:58
    #22 0x557a2effa70b in boost::unit_test::framework::run(unsigned long, bool) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/framework.ipp:1696:29
    #23 0x557a2f096446 in boost::unit_test::unit_test_main(boost::unit_test::test_suite* (*)(int, char**), int, char**) /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/unit_test_main.ipp:248:9
    #24 0x557a2f097e10 in main /tmp/cirrus-ci-build/depends/work/build/x86_64-pc-linux-gnu/boost/1_71_0-ae05fff5eee/./boost/test/impl/unit_test_main.ipp:304:12
    #25 0x7f79e95570b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #26 0x557a2c8b01ed in _start (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/test_bitcoin+0x1021ed)

  Uninitialized value was created by a heap allocation
    #0 0x557a2c8bc4ad in malloc (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/test_bitcoin+0x10e4ad)
    #1 0x557a2f2d3032 in sqlite3MemMalloc (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/test_bitcoin+0x2b25032)

SUMMARY: MemorySanitizer: use-of-uninitialized-value (/tmp/cirrus-ci-build/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/test/test_bitcoin+0x2a60abc) in sqlite3BtreeOpen
Exiting
make[3]: *** [Makefile:17263: wallet/test/psbt_wallet_tests.cpp.test] Error 1
```

--- a/ci/test/00_setup_env_native_fuzz_with_msan.sh
+++ b/ci/test/00_setup_env_native_fuzz_with_msan.sh
@@ -14,7 +14,7 @@ export MSAN_AND_LIBCXX_FLAGS="${MSAN_FLAGS} ${LIBCXX_FLAGS}"
 
 export CONTAINER_NAME="ci_native_msan"
 export PACKAGES="clang-9 llvm-9 cmake"
-export DEP_OPTS="NO_BDB=1 NO_QT=1 CC='clang' CXX='clang++' CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}' boost_cxxflags='-std=c++17 -fvisibility=hidden -fPIC ${MSAN_AND_LIBCXX_FLAGS}' zeromq_cxxflags='-std=c++17 ${MSAN_AND_LIBCXX_FLAGS}'"
+export DEP_OPTS="NO_BDB=1 NO_QT=1 CC='clang' CXX='clang++' CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}' boost_cxxflags='-std=c++17 -fvisibility=hidden -fPIC ${MSAN_AND_LIBCXX_FLAGS}' libevent_cflags='${MSAN_FLAGS}' zeromq_cxxflags='-std=c++17 ${MSAN_AND_LIBCXX_FLAGS}'"
 export GOAL="install"
 export BITCOIN_CONFIG="--enable-fuzz --with-sanitizers=fuzzer,memory --with-asm=no --prefix=${DEPENDS_DIR}/x86_64-pc-linux-gnu/ CC=clang CXX=clang++ CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}'"
 export USE_MEMORY_SANITIZER="true"

--- a/ci/test/00_setup_env_native_msan.sh
+++ b/ci/test/00_setup_env_native_msan.sh
@@ -15,7 +15,7 @@ export BDB_PREFIX="${BASE_ROOT_DIR}/db4"
 
 export CONTAINER_NAME="ci_native_msan"
 export PACKAGES="clang-9 llvm-9 cmake"
-export DEP_OPTS="NO_BDB=1 NO_QT=1 CC='clang' CXX='clang++' CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}' boost_cxxflags='-std=c++17 -fvisibility=hidden -fPIC ${MSAN_AND_LIBCXX_FLAGS}' zeromq_cxxflags='-std=c++17 ${MSAN_AND_LIBCXX_FLAGS}'"
+export DEP_OPTS="NO_BDB=1 NO_QT=1 CC='clang' CXX='clang++' CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}' boost_cxxflags='-std=c++17 -fvisibility=hidden -fPIC ${MSAN_AND_LIBCXX_FLAGS}' libevent_cflags='${MSAN_FLAGS}' sqlite_cflags='${MSAN_FLAGS}' zeromq_cxxflags='-std=c++17 ${MSAN_AND_LIBCXX_FLAGS}'"
 export GOAL="install"
 export BITCOIN_CONFIG="--enable-wallet --with-sanitizers=memory --with-asm=no --prefix=${DEPENDS_DIR}/x86_64-pc-linux-gnu/ CC=clang CXX=clang++ CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}' BDB_LIBS='-L${BDB_PREFIX}/lib -ldb_cxx-4.8' BDB_CFLAGS='-I${BDB_PREFIX}/include'"
 export USE_MEMORY_SANITIZER="true"

